File: main/src/main/java/lavaplayer/source/stream/MpegTsM3uStreamAudioTrack.java
package lavaplayer.source.stream;

import lavaplayer.container.adts.AdtsAudioTrack;
import lavaplayer.container.mpegts.MpegTsElementaryInputStream;
import lavaplayer.container.mpegts.PesPacketInputStream;
import lavaplayer.track.AudioTrackInfo;
import lavaplayer.track.playback.LocalAudioTrackExecutor;

import java.io.InputStream;

import static lavaplayer.container.mpegts.MpegTsElementaryInputStream.ADTS_ELEMENTARY_STREAM;

public abstract class MpegTsM3uStreamAudioTrack extends M3uStreamAudioTrack {
    /**
     * @param trackInfo Track info
     */
    public MpegTsM3uStreamAudioTrack(AudioTrackInfo trackInfo) {
        super(trackInfo);
    }

    @Override
    protected void processJoinedStream(LocalAudioTrackExecutor localExecutor, InputStream stream) throws Exception {
        MpegTsElementaryInputStream elementaryInputStream = new MpegTsElementaryInputStream(stream, ADTS_ELEMENTARY_STREAM);
        PesPacketInputStream pesPacketInputStream = new PesPacketInputStream(elementaryInputStream);

        processDelegate(new AdtsAudioTrack(trackInfo, pesPacketInputStream), localExecutor);
    }
}

File: main/src/main/java/lavaplayer/source/stream/MpegTsM3uStreamAudioTrack.kt
package lavaplayer.source.stream

import lavaplayer.container.adts.AdtsAudioTrack
import lavaplayer.container.mpegts.MpegTsElementaryInputStream
import lavaplayer.container.mpegts.PesPacketInputStream
import lavaplayer.track.AudioTrackInfo
import lavaplayer.track.playback.LocalAudioTrackExecutor
import java.io.InputStream

/**
 * @param trackInfo Track info
 */
abstract class MpegTsM3uStreamAudioTrack(trackInfo: AudioTrackInfo) : M3uStreamAudioTrack(trackInfo) {
    @Throws(Exception::class)
    override fun processJoinedStream(localExecutor: LocalAudioTrackExecutor, stream: InputStream) {
        val elementaryInputStream = MpegTsElementaryInputStream(stream, MpegTsElementaryInputStream.ADTS_ELEMENTARY_STREAM)
        val pesPacketInputStream = PesPacketInputStream(elementaryInputStream)
        processDelegate(AdtsAudioTrack(info, pesPacketInputStream), localExecutor)
    }
}
